src/App.tsx

```diff
@@ -15,48 +16,120 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortTypes {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortTypes,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortTypes.ALPABET:
+        return a.localeCompare(b);
+      case SortTypes.LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortTypes.NONE);
```

Comment:
Add generic if you use custom non-primitive values.
```suggestion
  const [sortType, setSortType] = useState<SortTypes>(SortTypes.NONE);
```